soepy/python/shared/shared_auxiliary.py

```diff
@@ -26,20 +26,22 @@ def draw_disturbances(num_draws, shocks_cov, seed):
     return draws
 
 
-def calculate_utilities(attr_dict, educ_level, exp_p, exp_f, optim_paras, draws):
+def calculate_utilities(model_params, educ_level, exp_p, exp_f, optim_paras, draws):
```

Comment:
This appears to be a break in our logic as optim_paras is part of model_params and we want to pass in model_params as deep into the code as we can. If this is a valid concern, then please check for this everywhere in the code as I see it happening repeatedly.